libs/s25main/PickedMovableObject.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2005 - 2023 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "DrawPoint.h"
+#include "gameTypes/MapCoordinates.h"
+
+class GameWorldView;
+
+class PickedMovableObject
+{
+public:
+    static PickedMovableObject Pick(const GameWorldView& gwv, MapPoint mapPt, DrawPoint drawPt, bool expire);
+    static PickedMovableObject PickAtCursor(const GameWorldView& gwv, bool expire);
+    static PickedMovableObject PickAtViewCenter(const GameWorldView& gwv, bool expire);
+
+    PickedMovableObject() = default;
+
+    // disable implicit copy; may still be useful
+    explicit PickedMovableObject(const PickedMovableObject&) = default;
+    PickedMovableObject& operator=(const PickedMovableObject&) = default;
+
+    // invalidates the moved-from object
+    PickedMovableObject(PickedMovableObject&& other) noexcept;
+    PickedMovableObject& operator=(PickedMovableObject&& other) noexcept;
+
+    unsigned Id() const { return id_; }
+    bool IsValid() const;
+
+    void CancelExpiration();
+    void Invalidate();
+
+    bool Track(const GameWorldView& gwv);
+    bool Track(GameWorldView& gwv, bool moveTo);
+
+private:
+    unsigned id_ = 0;
+    MapPoint mapPt_{};
+    DrawPoint drawPt_{};
+    unsigned long expiration_ = static_cast<unsigned long>(-1) /* = ExpireNever */;
```

Comment:
Might be useful to use a `Timer` instance here instead where "not running" means: Never expire   
This allows to use a `std::chrono` constant making the time unit explicit and testing the expiration mechanism if we wanted to through `MockClock`

Reply:
Will look into it.